### ISSUE TITLE
Join multiple paths

### ISSUE THREAD

**Issue by lunabunn** [RazrFalcon/tiny-skia#66]
I am looking to join multiple closed paths together so that they can be drawn together. It seems that `tiny-skia` does not provide a simple way to do this, and one must therefore resort to manually iterating over path segments and using `PathBuilder`.

Is this indeed the case? If so, perhaps something like `Path::join` or `PathBuilder::push_path` could be added? If not, what am I missing?

**Comment by RazrFalcon** [comment]
Yes, we should make `PathBuilder::push_path` public. And we can add a similar method to `Path` as well.

**Comment by lunabunn** [comment]
> Yes, we should make `PathBuilder::push_path` public. And we can add a similar method to `Path` as well.

The current private `PathBuilder::push_path` method actually takes a `PathBuilder` (and not a `Path`), which I am not sure would be very useful in a public API.

Ideally, the current `PathBuilder::push_path` should be renamed to something like `PathBuilder::push_path_builder` and kept private. A new public `PathBuilder::push_path` could then be added to do the same thing but with an actual `Path`. As you said, I think we can add a `pub fn join(self, other: Path) -> Path` method to `Path` as well.

If you'd be cool with something along these lines, I can create a PR and we can bikeshed from there (as necessary)?

**Comment by RazrFalcon** [comment]
Sure, sounds good.

**Comment by lunabunn** [comment]
Just an update: I don't actually need this anymore in my project and I have limited time lately, so I cannot make any guarantees as to when that PR is coming :(

I will probably get to it at some point. In the meantime, if anyone else wants to work on it, please feel more than free to do so unannounced.

**Comment by RazrFalcon** [comment]
No problem. I will add it eventually, when I would have time.
